howtos/diffs/distributed-training.diff

```diff
@@ -62,21 +62,21 @@ index 5dc74db..77b49fa 100644
    train_ds_size = len(train_ds['image'])
    steps_per_epoch = train_ds_size // batch_size
  
-@@ -141,6 +155,7 @@ def train_epoch(optimizer, train_ds, batch_size, epoch, rng):
+@@ -140,6 +154,7 @@ def train_epoch(optimizer, train_ds, batch_size, epoch, rng):
    batch_metrics = []
    for perm in perms:
      batch = {k: v[perm] for k, v in train_ds.items()}
 +    batch = shard(batch)
      optimizer, metrics = train_step(optimizer, batch)
      batch_metrics.append(metrics)
  
-@@ -192,7 +207,8 @@ def train(train_ds, test_ds):
-   for epoch in range(1, num_epochs + 1):
+@@ -191,7 +206,8 @@ def train(train_ds, test_ds):
+     rng, input_rng = random.split(rng)
      optimizer, train_metrics = train_epoch(
          optimizer, train_ds, batch_size, epoch, input_rng)
 -    loss, accuracy = eval_model(optimizer.target, test_ds)
 +    model = jax_utils.unreplicate(optimizer.target)  # Fetch from 1st device
-+    loss, accuracy = eval_model(model, test_ds)
++    loss_accuracy = eval_model(model, test_ds)
```

Comment:
Is this correct? Seem like a typo maybe? (Do HOWTO tests pass here? -- looks like yes, but maybe the tests don't yet cover this step?)

Reply:
Yes, this is a typo! The test tests `train_step`, but not `train`.